extensions/amp-consent/0.1/test/test-consent-info.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  CONSENT_ITEM_STATE,
+  getStoredConsentInfo,
+} from '../consent-info';
+import {dict} from '../../../../src/utils/object';
+
+
+describes.fakeWin('ConsentConfig', {}, () => {
+  describe('getStoredConsentInfo', () => {
+    it('construct consentInfo from undefined', () => {
+      expect(getStoredConsentInfo(undefined)).to.deep.equal(dict({
+        'consentState': CONSENT_ITEM_STATE.UNKNOWN,
+        'consentString': undefined,
+        'isDirty': undefined,
+      }));
+    });
+
+    it('construct consentInfo from legacy value', () => {
+      expect(getStoredConsentInfo(true)).to.deep.equal(dict({
+        'consentState': CONSENT_ITEM_STATE.ACCEPTED,
+        'consentString': undefined,
+        'isDirty': undefined,
+      }));
+      expect(getStoredConsentInfo(false)).to.deep.equal(dict({
+        'consentState': CONSENT_ITEM_STATE.REJECTED,
+        'consentString': undefined,
+        'isDirty': undefined,
+      }));
+    });
+
+    it('throw error with invalid value', () => {
+      expect(() => getStoredConsentInfo({})).to.throw(
+          'Invalid stored consent value');
+    });
+  });
+});
```

Comment:
Do we have a e2e test with this new struct? I assume it will break now?

Reply:
Can I know what kind of e2e test do you have in mind? I did test manually but didn't have an integration test for that. 

This should never happen, otherwise it is a dev error that we should fix. When the error is thrown, it will be treated consent state unknown. https://github.com/ampproject/amphtml/blob/ef1c5950bd607a5d441818a74c1baa10cece94ab/extensions/amp-consent/0.1/consent-state-manager.js#L274-L277

Reply:
Sorry if I was unclear, I didn't mean anything about this line. I meant a test somewhere else where you are writing then reading this cookie, but it is still in the old format.

Reply:
Ah, I thought the new format had already been implemented elsewhere, thanks for the clarification.